soweego/linker/feature_extraction.py

```diff
@@ -327,3 +328,123 @@ def intersection_percentage_size(pair):
+
+            # if we don't have them then we get them from
+            # wikidata and add them to the cache and
+            # `expanded_set`
+            else:
+
+                # get subclasses and superclasses
+                subclasses = sparql_queries.get_subclasses_of_qid(qid)
+                superclasses = sparql_queries.get_superclasses_of_qid(qid)
+
+                joined = subclasses | superclasses
+
+                # add joined to cache
+                self._expand_occupations_cache[qid] = joined
+
+                # finally add them to expanded set
+                expanded_set = expanded_set | joined
+
+        return expanded_set
+
+    def _compute_vectorized(self, source_column: pd.Series, target_column: pd.Series):
+
+        # each item in `source_column` is already an array composed
+        # of QID strings representing the occupations of a person.
+        # On the other hand, each item in `target_column` is a
+        # string of space separated QIDs, so we need to first convert
+        # those into lists
+        # then we also convert each item both the source and target
+        # from arrays to sets, so that they're easier to compare
+        
+        def to_set(itm):
+            # if it is an empty array (from source), or an
+            # empty string (from target)
+            if len(itm) == 0:
+                return set()
+
+            # if it is a string with length > 0 split it into
+            # its components
+            elif isinstance(itm, str):
+                itm = itm.split(" ")
+
+            return set(itm)
+        
+        target_column = target_column.apply(to_set)
+        source_column = source_column.apply(to_set)
+
+        # finally, we want to expand the target_column (add the
+        # superclasses and subclasses of each occupation)
+        target_column = target_column.apply(self._expand_occupations)
+
+        # we then zip together the source column and the target column so that
+        # they're easier to process
+        concatenated = pd.Series(list(zip(source_column, target_column)))
+
+        def check_occupation_equality(pair: Tuple[Set[str], Set[str]]):
+            """Given 2 sets, returns the percentage of items that the
+            smallest set shares with the larger set"""
+            s_item: Set
+            t_item: Set
+            s_item, t_item = pair
+
+            # explicitly check if set is empty
+            if s_item == set():
+                LOGGER.debug(
+                    "Can't compare occupations, the wikidata value is null. Pair: %s", pair)
+                return np.nan
+
+            if t_item == set():
+                LOGGER.debug(
+                    "Can't compare occupations, the target value is null. Pair: %s", pair)
+                return np.nan
+
+            smaller_length = min(len(s_item), len(t_item))
+            n_shared_items = len(s_item.intersection(t_item))
```

Comment:
```suggestion
            n_shared_items = len(s_item & t_item)
```
Just because you used boolean operators above